### DISCUSSION TITLE
Horse.Mime

### DISCUSSION THREAD

**Discussion by AciZuchinali** [HashLoad/horse#342] [General]
Boa tarde.
Estou com um problema ao compilar meu projeto.
Ao declarar a unit Horse ao projeto ocorre um erro de compilação onde o delphi não reconhece a unit Horse.Mime.

" var
    LKind: TMimeTypes.TKind;

[dcc32 Error] Horse.Mime.pas(67): E2003 Undeclared identifier: 'TMimeTypes'"

Grato.

**Comment by viniciussanchez** [comment]
Está usando a última versão disponível do Horse?
Qual a versão do seu Delphi?

**Reply by AciZuchinali** [reply]
Sim, estou usando a última versão, com delphi 10.2.

**Reply by illison** [reply]
Estou com a mesma situação

**Reply by viniciussanchez** [reply]
Nos envie um projeto pequeno com exemplo... aparentemente ninguém mais reclamou sobre isso nem mesmo nos grupos de Telegram e WhatsApp.

**Reply by vipeol** [reply]
Eu já tive este problema era a verão do delphi, como eu não poderia atualizar o Delphi tive que voltar a versão do middleware server-static para a 1.0

**Reply by AciZuchinali** [reply]
Eu comentei a linha e resolveu.

**Comment by AlexBertoldo** [comment]
Olá,  Estou com o mesmo problema ao compilar o exemplo "Quickstart Delphi" do horse
Estou usando o Delphi 10.2 Tokio. E esta dando erro ao compilar Horse.Mime.

[dcc32 Error] Horse.Mime.pas(67): E2003 Undeclared identifier: 'TMimeTypes'
[dcc32 Error] Horse.Mime.pas(67): E2029 ',' or ':' expected but ';' found
[dcc32 Error] Horse.Mime.pas(69): E2029 Type expected but 'BEGIN' found
[dcc32 Error] Horse.Mime.pas(71): E2003 Undeclared identifier: 'TMimeTypes'
[dcc32 Error] Horse.Mime.pas(71): E2066 Missing operator or semicolon
[dcc32 Error] Horse.Mime.pas(71): E2029 ')' expected but ';' found
[dcc32 Error] Horse.Mime.pas(80): E2003 Undeclared identifier: 'TMimeTypes'
[dcc32 Error] Horse.Mime.pas(80): E2029 ',' or ':' expected but ';' found
[dcc32 Error] Horse.Mime.pas(82): E2029 Type expected but 'BEGIN' found
[dcc32 Error] Horse.Mime.pas(84): E2003 Undeclared identifier: 'TMimeTypes'
[dcc32 Error] Horse.Mime.pas(84): E2066 Missing operator or semicolon
[dcc32 Error] Horse.Mime.pas(84): E2029 ')' expected but ';' found
[dcc32 Fatal Error] Horse.Core.Files.pas(40): F2063 Could not compile used unit 'Horse.Mime.pas'

**Comment by AlexBertoldo** [comment]
Funcionou
Obrigado
Deus Abençoe

Em ter., 26 de dez. de 2023 às 15:29, Aci Zuchinali <
***@***.***> escreveu:

> Eu comentei a linha e resolveu.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HashLoad/horse/discussions/342#discussioncomment-7950699>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZKKO2TB37EFEPIH7TD4SGTYLMJRXAVCNFSM6AAAAAAXUUQSUWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSNJQGY4TS>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Alex Bertoldo*
*Analista de Sistemas*
Whats 11 973897627

**Comment by rcloayza** [comment]
Me sucede el mismo error, tengo Delphi 10.2

Al compilar se produce el siguiente error:
[dcc32 Error] Horse.Mime.pas(68): E2003 Undeclared identifier: 'TMimeTypes'

class function THorseMimeTypes.GetExtType(const AFileExt: string): string;
  {$IFNDEF UseTHorseMimeTypesExt}
  var
    **LKind: TMimeTypes.TKind;** //ERROR
  {$ENDIF}
begin
  {$IFNDEF UseTHorseMimeTypesExt}
  System.Net.Mime.TMimeTypes.Default.GetExtInfo(ExtractFileExt(AFileExt), Result, LKind);
  {$ELSE}
  Result := THorseMimeTypesExt.Default.GetExtType(AFileExt);
  {$ENDIF}
end;

Realice el siguiente cambio en la directiva de compilación
_{.$IF DEFINED(FPC) OR (CompilerVersion < 32.0)}_
{$IF DEFINED(FPC) OR (CompilerVersion < **33.0**)} **//SIN ERROR DE COMPILACION...**

Esto es debido a que en la unit **System.Net.Mime** no existe  System.Net.Mime.TMimeTypes

Saludos cordiales

**Reply by viniciussanchez** [reply]
Corrigido no commit abf8c10e1d84475682fa59ad74a382cac9a8d616
